web/libs/editor/src/tags/control/VectorLabels.jsx

```diff
@@ -0,0 +1,87 @@
+ * @param {string} name                             - Name of tag
+ * @param {string} toName                           - Name of image to label
+ * @param {single|multiple=} [choice=single]        - Configure whether you can select one or multiple labels
+ * @param {number} [maxUsages]                      - Maximum number of times a label can be used per task
+ * @param {boolean} [showInline=true]               - Show labels in the same visual line
+ * @param {number} [opacity=0.2]                    - Opacity of vector
+ * @param {string} [fillColor]                      - Vector fill color in hexadecimal
+ * @param {string} [strokeColor]                    - Stroke color in hexadecimal
+ * @param {number} [strokeWidth=1]                  - Width of stroke
+ * @param {small|medium|large} [pointSize=medium]   - Size of vector handle points
+ * @param {rectangle|circle} [pointStyle=rectangle] - Style of points
+ * @param {pixel|none} [snap=none]                  - Snap vector to image pixels
+ * @params {boolean} [closable=false]               - Allow closed shapes
+ * @params {boolean} [curves=false]                 - Allow Bezier curves
+ * @params {boolean} [skeleton=false]               - Enables skeleton mode to allow branch paths
+ * @params {number|none} [minPoints=none]           - Minimum allowed number of points
+ * @params {number|none} [maxPoints=none]           - Maximum allowed number of points
+ * @params {boolean} [constrainToBounds=false]      - Whether to keep shapes inside image bounds
+ * @params {number} [pointnSizeEnabled=5]           - Size of a point in pixels when shape is selected
+ * @params {number} [pointnSizeDisabled=5]          - Size of a point in pixels when shape is not selected
+ */
+
+const Validation = types.model({
+  controlledTags: Types.unionTag(["Image"]),
+});
+
+const ModelAttrs = types.model("VectorLabelsModel", {
+  type: "vectorlabels",
+  closable: types.optional(types.maybeNull(types.boolean), false),
+  curves: types.optional(types.maybeNull(types.boolean), false),
+  minpoints: types.optional(types.maybeNull(types.string), null),
+  maxpoints: types.optional(types.maybeNull(types.string), null),
+  constraintobounds: types.optional(types.maybeNull(types.boolean), false),
+  skeleton: types.optional(types.maybeNull(types.boolean), false),
+  pointnsizeenabled: types.optional(types.maybeNull(types.string), "5"),
+  pointnsizedisabled: types.optional(types.maybeNull(types.string), "3"),
+  opacity: types.optional(types.maybeNull(types.string), "1"),
+  children: Types.unionArray(["label", "vectorlabel", "header", "view", "hypertext"]),
+});
+
+const Composition = types.compose(
+  ControlBase,
+  LabelsModel,
+  ModelAttrs,
+  VectorModel,
+  Validation,
+  LabelMixin,
+  SelectedModelMixin.props({ _child: "LabelModel" }),
+);
+
+const VectorLabelsModel = types.compose("VectorLabelsModel", Composition);
```

Comment:
small inconsistency: MST supports and we usually use simple composition with name along with no extra steps:
```
const VectorLabelsModel = types.compose("VectorLabelsModel", ControlBase, LabelsModel, ...)
```